zebra-chain/src/sapling/arbitrary.rs

```diff
@@ -99,3 +100,49 @@ impl Arbitrary for OutputInTransactionV4 {
 
     type Strategy = BoxedStrategy<Self>;
 }
+
+/// A wrapper for a RedJubJub spending verification key.
+///
+/// This is used by proptests since we can't implement Arbitrary for the
+/// redjubjub::VerificationKeyBytes (it's in another crate).
+/// We then implement it for the wrapper type instead.
+#[derive(Debug)]
+struct SpendVerificationKeyBytesWrapper(redjubjub::VerificationKeyBytes<redjubjub::SpendAuth>);
+
+impl Arbitrary for SpendVerificationKeyBytesWrapper {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (vec(any::<u8>(), 32))
+            .prop_filter_map("invalid verification key", |bytes| {
+                let bytes: [u8; 32] = bytes.try_into().expect("vec is the correct length");
+                let vkb = redjubjub::VerificationKeyBytes::<redjubjub::SpendAuth>::try_from(bytes)
+                    .expect("a valid generated verification key bytes");
+                // Convert to a VerificationKey to make sure it's valid;
+                // but return the underlying bytes if it works.
+                let r = redjubjub::VerificationKey::<redjubjub::SpendAuth>::try_from(vkb);
+                r.ok().map(|_| SpendVerificationKeyBytesWrapper(vkb))
+            })
+            .boxed()
+    }
+
+    type Strategy = BoxedStrategy<Self>;
+}
+
+impl Arbitrary for tree::Root {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (vec(any::<u8>(), 32))
+            .prop_filter_map("invalid Sapling nore commitment tree root", |bytes| {
+                let bytes = bytes.try_into().expect("vec is the correct length");
+                // Convert to a Scalar to make sure it's valid;
+                // but return the underlying bytes if it works.
+                let r = bls12_381::Scalar::from_repr(bytes);
```

Comment:
Consistency: is there a `from_bytes_wide` for `bls12_381::Scalar`, or another method that takes arbitrary bytes, and always successfully creates a scalar? That seems like the best option here.

If not, how frequent are failures?

The docs say:
> should only be used if relatively few values will actually be rejected

But they don't give any specific guidelines.

So I think it's ok to document the failure rate, open a follow-up ticket, and merge as is.